From d1877c5771b11fcd5123d7ef4e1d2b88cb980be9 Mon Sep 17 00:00:00 2001
From: Sihyun Baek <62822977+sihyunlts@users.noreply.github.com>
Date: Fri, 16 Oct 2026 16:28:20 +0900
Subject: [PATCH] fs_mgr: enable automatic F2FS checks during boot

Pass -a to the CM11 fsck.f2fs tool so detected errors can be handled without an interactive prompt during boot.

Backport only the fs_mgr option change from AOSP commit 784c22c8388e50db243ac4ca3871747cd3eefadc. Its init, reboot and shutdown-scan changes are outside this backport.

Change-Id: Ia565bd1a07bd8720f09d6563660c80e81d7ae95c
---
 fs_mgr/fs_mgr.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/fs_mgr/fs_mgr.c b/fs_mgr/fs_mgr.c
index 46d1f7e..65e23c4 100644
--- a/fs_mgr/fs_mgr.c
+++ b/fs_mgr/fs_mgr.c
@@ -449,9 +449,10 @@ static void check_fs(char *blk_device, char *fs_type, char *target)
     } else if (!strcmp(fs_type, "f2fs")) {
             char *f2fs_fsck_argv[] = {
                     F2FS_FSCK_BIN,
+                    "-a",
                     blk_device
             };
-        INFO("Running %s on %s\n", F2FS_FSCK_BIN, blk_device);
+        INFO("Running %s -a %s\n", F2FS_FSCK_BIN, blk_device);
 
         ret = android_fork_execvp_ext(ARRAY_SIZE(f2fs_fsck_argv), f2fs_fsck_argv,
                                       &status, true, LOG_KLOG | LOG_FILE,
